### PR TITLE
Use ihaskell flake

### PR DESCRIPTION
**Pull request by teto**
I've added a flake to ihaskell recently. We can fetch the ihaskell overlay from that flake to avoid code duplication.

### CODE REVIEW

**Comment by teto** [comment]
@guaraqe this is ready for review. Not sure what the macos error is about

**Comment by guaraqe** [comment]
Can you rebase on master? I updated the MacOS action, which should help solve the error.

File: example/flake/flake.nix [hunk 1]
```diff
@@ -7,7 +7,7 @@
   inputs = {
     flake-utils.url = "github:numtide/flake-utils";
     nixpkgs.url = "nixpkgs/a5d03577f0161c8a6e713b928ca44d9b3feb2c37";
-    jupyterWith = { url = "path:../../."; };
+    jupyterWith.url = github:tweag/jupyterWith;
```
Comment (teto) [inline example/flake/flake.nix:10]: had to change it else when starting the flake from the example/flake folder it gives `relative path '../../.' points outside of its parent's store path '/nix/store/j6hpzfy2zi7pm8pxv1l61h1j4v24mpfc-source'`

**Comment by teto** [comment]
I've rebased but still have the issue sadly. I dont know about macs, feel free to commit here if htis can help getting this merged. I am working on improving ihaskell and having the ability to override the ihaskell version of jupyterWith via the command line is helpful in debugging.

**Comment by guaraqe** [comment]
> looks like the python was built by two different versions(python3.8 python3.9). therefore, Python 3.8 would be ok on macOS.

We have a mention to it in another issue. Are you able to locate the source of these two different Pythons? Maybe just changing all the `python` with `python38` would do the job.

**Pull request merged**